fl4health/checkpointing/checkpointer.py

```diff
@@ -2,71 +2,157 @@
+                return self.best_score <= comparison_score
             else:
-                return self.best_metric >= comparison_metric
+                return self.best_score >= comparison_score
 
-        # If best metric is none, then this is the first checkpoint
+        # If best score is none, then this is the first checkpoint
         return True
 
-    def maybe_checkpoint(self, model: nn.Module, comparison_metric: Optional[float] = None) -> None:
-        assert comparison_metric is not None
-        if self._should_checkpoint(comparison_metric):
+    def maybe_checkpoint(self, model: nn.Module, loss: float, metrics: Dict[str, Scalar]) -> None:
+        # First we use the provided scoring function to produce a score
+        comparison_score = self.checkpoint_score_function(loss, metrics)
+        if self._should_checkpoint(comparison_score):
+            log(
+                INFO,
+                f"Checkpointing the model: Current score ({comparison_score}) "
+                f"{self.comparison_str} Best score ({self.best_score})",
+            )
+            self.best_score = comparison_score
+            torch.save(model, self.best_checkpoint_path)
+        else:
+            log(
+                INFO,
+                f"Not checkpointing the model: Current score ({comparison_score}) is not "
+                f"{self.comparison_str} Best score ({self.best_score})",
+            )
+
+
+class LatestTorchCheckpointer(TorchCheckpointer):
+    def __init__(self, checkpoint_dir: str, checkpoint_name: str) -> None:
+        super().__init__(checkpoint_dir, checkpoint_name)
+
+    def maybe_checkpoint(
+        self, model: nn.Module, loss: Optional[float] = None, metrics: Optional[Dict[str, Scalar]] = None
+    ) -> None:
+        # Always checkpoint the latest model
+        log(INFO, "Saving latest checkpoint with LatestTorchCheckpointer")
+        torch.save(model, self.best_checkpoint_path)
+
+
+def loss_score_function(loss: float, _: Dict[str, Scalar]) -> float:
+    return loss
+
+
+class BestLossTorchCheckpointer(FunctionTorchCheckpointer):
+    def __init__(self, checkpoint_dir: str, checkpoint_name: str) -> None:
+        """
+        This checkpointer only uses the loss value provided to the maybe_checkpoint function to determine whether a
+        checkpoint should be save. We are always attempting to minimize the loss. So maximize is always set to false
+
+        Args:
+            checkpoint_dir (str): Directory to which the model is saved. This directory should already exist. The
+                checkpointer will not create it if it does not.
+            checkpoint_name (str): Name of the checkpoint to be saved.
+        """
+        super().__init__(
+            checkpoint_dir, checkpoint_name, checkpoint_score_function=loss_score_function, maximize=False
+        )
+
+    def maybe_checkpoint(self, model: nn.Module, loss: float, metrics: Dict[str, Scalar]) -> None:
```

Comment:
Note, I'm overriding this method to replace the logging with something more specific. If anyone has any better ideas on how to do this, let me know.